eventhubs-streamanalytics-eventhubs/tools/eh-asa-perfmon/EventHandler.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Azure.Messaging.EventHubs.Consumer;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace StreamingAtScale
+{
+    public class EventReceiver
+    {
+        private readonly object _lock = new object();
+
+        public async Task Receive(string connectionString, CancellationToken _cancellationToken)
+        {
+            await using (var consumerClient = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, connectionString))
+            {
+                Console.WriteLine("The application will now start to listen for incoming message.");
+
+                ReadEventOptions readOptions = new ReadEventOptions
+                {
+                    MaximumWaitTime = TimeSpan.FromMilliseconds(150)
+                };
+
+                var csvOutput = File.CreateText("./resulttest.csv");
+                csvOutput.WriteLine("EventCount,BatchCount,BatchFrom,BatchTo,MinLatency,MaxLatency,AvgLatency");
+                try
+                {
+                    var events = consumerClient.ReadEventsAsync(startReadingAtEarliestEvent: false, readOptions, _cancellationToken);
```

Comment:
It is probably worth adding some comments to note that the client is safe (and intended) to be long-lived.  If this class was to be used for a while and `Receive` called over-and-over, the client would be better as a class member.   That's not the case with this sample, so it's fine to leave the code as-is, but this is an area of confusion that comments would help clarify.

Reply:
resolved